ext/standard/tests/class_object/get_declared_enums_basic.phpt

```diff
@@ -0,0 +1,66 @@
+--TEST--
+Test get_declared_enums() function
+--FILE--
+<?php
+
+enum MyUnitEnum {}
+enum MyStringBackedEnum: string {
+    case FOO = 'test';
+}
+enum MyIntBackedEnum: int {
+    case FOO = 1;
+}
+enum MyEnumImplementation implements Countable {
+    public function count(): int {
+        return 0;
+    }
+}
+
+var_dump(get_declared_enums());
+
+foreach (get_declared_enums() as $enum) {
+    if (!enum_exists($enum)) {
+        echo "Error: $enum is not a valid enum.\n";
+    }
+}
+
+echo "Ensure all enums are included.\n";
+
+var_dump(in_array('MyUnitEnum', get_declared_enums()));
```

Comment:
Considering Mike's remark on the mailing list about the hash table creation, would calling `get_declared_enums()` once, storing the result to a variable and doing the `in_array()` checks against the variable make the test slightly more efficient ?
(same for the calls to `get_declared_classes()`)

Reply:
Yeah I think it makes sense. `get_declared_enums()` took about 0.46 seconds for a million iterations, so the impact would be quite minimal. But I did follow your idea and added variables for `get_declared_(enums|classes)`.